pkg/v1/hash_test.go

```diff
@@ -83,3 +83,28 @@ func TestSHA256(t *testing.T) {
 		t.Errorf("n; got %v, want %v", got, want)
 	}
 }
+
+// This tests that you can use Hash as a key in a map (needs to implement both
+// MarshalText and UnmarshalText).
+func TestTextMarshalling(t *testing.T) {
+	foo := make(map[Hash]string)
```

Comment:
Is this JSON-marshalling test more than an elaborate type assertion, checking that it satisfies the `TextMarshaler` interface?

Reply:
This one is weird to me. Originally, I just had this json marshalling code in the test, which would fail unless I implemented the text marshalling. Unfortunately, the json marshalling doesn't actually provide coverage for the text marshalling implementation (I don't understand why), so I added the second half of this test to grab the coverage.